modules/bluefish/consumer/bluefish_consumer.cpp

```diff
@@ -75,7 +133,14 @@ struct bluefish_consumer : boost::noncopyable
 			
 	unsigned int										vid_fmt_;
 
-	std::array<blue_dma_buffer_ptr, 4>					reserved_frames_;	
+	std::array<blue_dma_buffer_ptr, 4>					all_frames_;	
+	std::queue<blue_dma_buffer_ptr>						reserved_frames_;
+	std::mutex											reserved_frames_lock_;
+	std::queue<blue_dma_buffer_ptr>						live_frames_;
+	std::mutex											live_frames_lock_;
+	std::shared_ptr<std::thread>						dma_present_thread_;
+	bool												end_dma_thread_;
```

Comment:
This needs to be a `tbb::atomic<bool>` to be able to be guarantee that the change from `false` to `true` is memory visible by the DMA thread, otherwise a cached value of `false` could be read.